crates/snapshot/src/snapshotter.rs

```diff
@@ -18,30 +18,21 @@ pub type SnapshotterWithResult<DB> = (Snapshotter<DB>, SnapshotterResult);
 /// Snapshotting routine. Main snapshotting logic happens in [Snapshotter::run].
 #[derive(Debug)]
 pub struct Snapshotter<DB> {
+    /// Provider factory
     provider_factory: ProviderFactory<DB>,
+    /// Highest snapshotted block numbers for each segment
     highest_snapshots: HighestSnapshots,
-    highest_snapshots_tracker: watch::Sender<Option<HighestSnapshots>>,
+    /// Channel sender to notify other components of the new highest snapshots
+    highest_snapshots_notifier: watch::Sender<Option<HighestSnapshots>>,
```

Comment:
do we expect features that need to listen for changed snapshots?

if not perhaps we could just use an rwlock wrapper type? because a watch channel is also just an rwlock with some add ons.

if we need listeners for snapshots we can add this separately

Reply:
yeah, agree. i'll do it in a further PR if that's okay